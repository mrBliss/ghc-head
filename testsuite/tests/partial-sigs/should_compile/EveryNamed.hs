{-# LANGUAGE NamedWildcards #-}
module EveryNamed where

every :: (_a -> Bool) -> [_a] -> Bool
every _ [] = True
every p (x:xs) = p x && every p xs

test :: (Bool, Bool)
test = (every testChar "Foo", every testBool [True, False, True])
  where testChar :: Char -> Bool
        testChar = undefined
        testBool :: Bool -> Bool
        testBool = undefined
