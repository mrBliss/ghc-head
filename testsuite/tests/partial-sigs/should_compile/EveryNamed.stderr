TYPE SIGNATURES
  every :: forall w_a. (w_a -> Bool) -> [w_a] -> Bool
TYPE CONSTRUCTORS
COERCION AXIOMS
Dependent modules: []
Dependent packages: [base, ghc-prim, integer-gmp]
